### PR TITLE
Add inheritance diagram to mpl.ticker docs

### PR DESCRIPTION
**Pull request by dstansby**
Does what it says on the tin, it's a bit of a high inheritance diagram so I've put it under the main docs. Also adds a couple of missing entries to `__all__`.

### CODE REVIEW

**Comment by anntzer** [comment]
https://24496-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/ticker_api.html looks empty...

File: doc/api/ticker_api.rst [hunk 1]
```diff
@@ -6,3 +6,7 @@
    :members:
    :undoc-members:
    :show-inheritance:
+
+
+ .. inheritance-diagram:: matplotlib.ticker
```
Comment (tacaswell) [inline doc/api/ticker_api.rst]: I think that this needs to a list of the classes you want included in the diagram, not a module.

Comment (dstansby) [reply doc/api/ticker_api.rst]: Hmm, it works locally (and specifying the module works elsewhere in the docs), not sure what's going on here...

Comment (timhoffm) [reply doc/api/ticker_api.rst]: The build log contains this:

> WARNING: invalid signature for automodule ('matplotlib.ticker\n  :members:\n  :undoc-members:\n  :show-inheritance:')
WARNING: don't know which module to import for autodocumenting 'matplotlib.ticker\n  :members:\n  :undoc-members:\n  :show-inheritance:' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)

**Comment by dstansby** [comment]
I think there was an indentation error, *hopefully* should be fixed now

**Pull request merged**